__tests__/Unit/Components/Tasks/TaskDetails.test.tsx

```diff
@@ -1,41 +1,311 @@
-import { fireEvent, render, screen } from '@testing-library/react';
-import TaskDetails from '@/components/taskDetails';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import TaskDetails, { Button, Textarea } from '@/components/taskDetails';
 import TaskContainer from '@/components/taskDetails/TaskContainer';
 import task from '@/interfaces/task.type';
 import { tasks } from '../../../../__mocks__/db/tasks';
+import { Provider } from 'react-redux';
+import { store } from '@/app/store';
+import { renderWithRouter } from '@/test_utils/createMockRouter';
+import { setupServer } from 'msw/node';
+import handlers from '../../../../__mocks__/handlers';
+import { ButtonProps, TextAreaProps } from '@/interfaces/taskDetails.type';
+import { ToastContainer } from 'react-toastify';
+import Details from '@/components/taskDetails/Details';
 
 const details = {
     url: 'https://realdevsquad.com/tasks/6KhcLU3yr45dzjQIVm0J/details',
     taskID: '6KhcLU3yr45dzjQIVm0J',
 };
 
-describe.skip('TaskDetails Page', () => {
-    test('Loading text rendered when loading', () => {
-        render(<TaskDetails url={details.url} taskID={details.taskID} />);
-        const loadingElement = screen.getByText(/Loading.../i);
-        expect(loadingElement).toBeInTheDocument();
+const server = setupServer(...handlers);
+
+beforeAll(() => {
+    server.listen({
+        onUnhandledRequest: 'error',
     });
-    test('Task title is Editable in Editing mode ', () => {
-        render(<TaskDetails url={details.url} taskID={details.taskID} />);
+});
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
 
-        const titleElement = screen.queryByTestId('task-title');
-        expect(titleElement).not.toBeInTheDocument();
+jest.mock('@/hooks/useUserData', () => {
```

Comment:
Are we sure we want to mock the whole module? This statement mocks the return value of `@hooks/useUserData` and not the return value of `useUserData` hook defined in it

Reply:
I will mention my observation here 👍🏻